[ADD] CRM: Forecast report (#9582)
odoo/documentation: content/applications/sales/crm/performance/forecast_report.rst

.. |list| replace:: :icon:`oi-view-list` :guilabel:`(list)` icon

The *Forecast* report in the *CRM* app allows users to view upcoming opportunities and build a
forecast of potential sales. Opportunities are grouped by the month of their expected closing date,
and can be dragged-and-dropped to adjust the deadline.

To access the *Forecast* report, navigate to :menuselection:`CRM app --> Reporting --> Forecast`.

.. image:: forecast_report/reporting-menu.png
   :align: center
   :alt: The location of the Forecast report in the CRM app.

Navigate the forecast report
============================

The default forecast report includes opportunities assigned to the current user's pipeline, and
expected to close within four months, as well as opportunities without an assigned expected closing
date. The opportunities are grouped by month in a Kanban (:icon:`oi-view-kanban`) view.

.. image:: forecast_report/sample-report.png
   :align: center
   :alt: A sample version of the Forecast report in the CRM app.

Expected closing date
---------------------

Opportunities are grouped by the date assigned in the :guilabel:`Expected Closing` field. To change

> this date, select the Kanban card for the desired opportunity, then click and drag the card to the

Reviewer: ```suggestion
this date directly from the :guilabel:`Forecast` page, select the Kanban card for the desired opportunity, then click and drag the card to the
```
SUGGESTION: for added specificity and enhanced clarity

desired column.

.. note::
   The default time frame for the forecast is *month*. This can be changed by clicking the |caret|
   next to the :guilabel:`Search...` bar at the top of the report. Under the :guilabel:`Group By`
   heading, click :guilabel:`Expected Closing` to expand the list of available options, and select
   from the list.

> After an opportunity is added to a new month, the :guilabel:`Expected Closing` field is updated to

Reviewer: ```suggestion
After an opportunity is added to a new month, the *Expected Closing* field on the opportunity form is updated to
```
SUGGESTION: added specificity to fall in line with the same reasoning as above re: the addition of "opportunity form" to the explanation. (using italics because that's a field name that isn't being shown directly on the 'Forecast' page -- it's only present on the Opportunity form)

the *last* date of the new month.

.. tip::

>    The :guilabel:`Expected Closing` field can also be manually updated. Click on the Kanban card for
>    an opportunity to open the opportunity's details form. Click in the :guilabel:`Expected Closing`
>    field, and use the calendar popover to select a new closing date.

Reviewer: ```suggestion
   The *Expected Closing* field can also be manually updated on the opportunity card. To do that, click on the Kanban card for
   an opportunity on the :guilabel:`Forecast` page to open the opportunity's detail form. Click in the :guilabel:`Expected Closing`
   field, and use the calendar popover to select a new closing date.
```

Reviewer: OPTIONAL: adjusted formatting (when necessary) and added specificity to the steps for enhanced clarity

Prorated revenue
----------------

> At the top of the column for each month, to the right of the progress bar, is a sum of the prorated

Reviewer: ```suggestion
At the top of the column for each month on the :guilabel:`Forecast` reporting page, to the right of the progress bar, is a sum of the prorated
```
OPTIONAL: just a helpful reminder to the reader, so they can always make sure they're on the right page

revenue for that time frame.

The prorated revenue is calculated using the formula below:

.. math::

>    \text{Expected Revenue} \times \text{Probability} = \text{Prorated Revenue}

Reviewer 2: wht a brilliant workaround

As opportunities are moved from one column to another, the column's revenue is automatically updated
to reflect the change.

.. example::
   A forecast report for June includes two opportunities:

>    The first opportunity, `Global Solutions` has an expected revenue of `$3,800` and a probability

Reviewer: ```suggestion
   The first opportunity, `Global Solutions`, has an expected revenue of `$3,800`, and a probability
```
SUGEGSTION: added commas for additional sentence clarity and enhanced flow

   of `90%`. This results in a prorated revenue of `$3,420`.

>    The second opportunity, `Quote for 600 Chairs` has an expected revenue of `$22,500`, and a

Reviewer: ```suggestion
   The second opportunity, `Quote for 600 Chairs`, has an expected revenue of `$22,500`, and a
```

   probability of `20%`. This results in a prorated revenue of `$4,500`.

   The combined prorated revenue of the opportunities is `$7,920`, which is listed at the top of the
   column for the month.

   .. image:: forecast_report/example-revenue.png
      :align: center
      :alt: An example of the prorated revenue for one month of the forecast report.

.. seealso::
   For more information on how probability is assigned to opportunities, see
   :doc:`../track_leads/lead_scoring`

View results
============

> Click the :icon:`fa-area-chart` :guilabel:`(area chart)` icon to change to graph view. Click the

Reviewer: I was going to leave a very similar suggestion here as @larm-odoo -- but since you've already spoken to Sam about it -- you can just keep it as-is 👍

> Click the :icon:`fa-area-chart` :guilabel:`(area chart)` icon to change to graph view. Click the

Reviewer: ```suggestion
Click the :icon:`fa-area-chart` :guilabel:`(area chart)` icon to change to graph view. Then, click the
```

> Click the :icon:`fa-area-chart` :guilabel:`(area chart)` icon to change to graph view. Click the

Reviewer 2: I know we are supposed to use the icon name, but this really says 'Graph' and that is more clear to me. I woudl argue for the use of "graph" but of course, defer to SALI. I think we need to has this out in a writers talk since thisocmes up a lot, especially when the icon description is bizarre.

Author: I've spoken to Sam about this, we are going to use the official names for now. There are a few icons that have names that may not match directly with the action on the screen that in the future I could see changing the term for (like the save icon, since the official name is "cloud upload" I believe) but for now we are supposed to use the dev name for consistency.

corresponding icon at the top of the report to switch to a :icon:`fa-bar-chart` :guilabel:`(bar
chart)`, :icon:`fa-line-chart` :guilabel:`(line chart)`, or :icon:`fa-pie-chart` :guilabel:`(pie
chart)`.

.. image:: forecast_report/pie-chart-view.png
   :align: center
   :alt: A pie chart view of the Forecast report.

> Click the |pivot| to change to the *pivot* view, or the |list| to change to the *list* view.

Reviewer: ```suggestion
Click the |pivot| to change to the pivot view, or the |list| to change to the list view.
```
no italics (nor capitalization) necessary here. it's my understanding that the only view options that we change at all are Kanban and Gantt -- and even then, it's just capitalized. feel free to double-check with sam for clarification, if you'd like 👍

.. tip::

>    The pivot view can be used to view and analyze data in a more in-depth manner. Multiple measures

Reviewer: I would also suggest linking to a Reporting view specific doc to let the reader quickly access more detailed information about this kind of view -- maybe to just a specific anchor? Either way, it's something to consider...

>    can be selected, and data can be viewed by month, and by opportunity stage.

Reviewer 2: This is a good doc to link to if you see it fts: https://www.odoo.com/documentation/17.0/applications/essentials/search.html

Author: This doc has a lot for filters and groupings, but not specifically measures, which is what the tip is directing to

   .. image:: forecast_report/pivot-view.png
      :align: center
      :alt: A sample of the forecast report in the pivot view.

.. seealso::

>    To save this report as a *favorite*, see :ref:`search/favorites`.

Reviewer: ```suggestion
   To save this (or any) report configuration in the *Favorites* column of the :guilabel:`Search...` bar drop-down menu of filter and grouping options, see :ref:`search/favorites`.
```
OPTIONAL: additional specification to enhanced the clarity of this explanation...

===============
Forecast report
===============

.. |caret| replace:: :icon:`fa-caret-down` :guilabel:`(down)` icon
.. |pivot| replace:: :icon:`oi-view-pivot` :guilabel:`(pivot)` icon
.. |list| replace:: :icon:`oi-view-list` :guilabel:`(list)` icon

The *Forecast* report in the *CRM* app allows users to view upcoming opportunities and build a
forecast of potential sales. Opportunities are grouped by the month of their expected closing date,
and can be dragged-and-dropped to adjust the deadline.

To access the *Forecast* report, navigate to :menuselection:`CRM app --> Reporting --> Forecast`.

> .. image:: forecast_report/reporting-menu.png
>    :align: center
>    :alt: The location of the Forecast report in the CRM app.
> 

Reviewer 3: In my opinion we could do without this image, just to reduce the number of overall images :)
```suggestion
```

Navigate the forecast report
============================

The default :guilabel:`Forecast` report includes opportunities assigned to the current user's
pipeline, and are expected to close within four months. It also shows opportunities without an

> assigned expected closing date. The opportunities are grouped by month in a Kanban
> (:icon:`oi-view-kanban`) view.

Reviewer 3: Just to stay consistent with how the other icons and views are presented in this doc
```suggestion
assigned expected closing date. The opportunities are grouped by month in a :icon:`oi-view-kanban`
:guilabel:`(Kanban)` view.
```
